src/Juvix/Compiler/Concrete/Print/Base.hs

```diff
@@ -59,23 +59,33 @@ ppModulePathType x = case sing :: SStage s of
 instance SingI t => PrettyPrint (Module 'Scoped t) where
   ppCode :: forall r. (Members '[ExactPrint, Reader Options] r) => Module 'Scoped t -> Sem r ()
   ppCode Module {..} = do
-    let moduleBody' = indent (ppCode _moduleBody)
+    let moduleBody' = localIndent (ppCode _moduleBody)
         modulePath' = ppModulePathType _modulePath
         moduleDoc' :: Sem r () = maybe (return ()) ppCode _moduleDoc
     moduleDoc'
       <> ppCode _moduleKw
       <+> modulePath'
         <> ppCode kwSemicolon
         <> line
+        <> topSpace
         <> moduleBody'
         <> line
-        <> ppCode kwEnd
-        <> lastSemicolon
+        <>? ending
     where
-      lastSemicolon :: Sem r ()
-      lastSemicolon = case sing :: SModuleIsTop t of
-        SModuleLocal -> return ()
-        SModuleTop -> semicolon <> line <> end
+      topSpace :: Sem r ()
+      topSpace = case sing :: SModuleIsTop t of
+        SModuleLocal -> mempty
+        SModuleTop -> line
+
+      localIndent :: Sem r () -> Sem r ()
+      localIndent = case sing :: SModuleIsTop t of
+        SModuleLocal -> indent
+        SModuleTop -> id
+
+      ending :: Maybe (Sem r ())
```

Comment:
You don't need `Maybe` here? Or are you trying to keep it the same as The `Pretty/Base.hs` version?

Reply:
Fixed. I forgot to remove the `Maybe`, thanks!